src/install.ts

```diff
@@ -68,3 +69,39 @@ export async function install(platform: string, architecture: string, release: s
 
     return;
 }
+
+// Limitations
+//
+// * No system dependencies are installed
+// * Does not cache
+export async function installFromSource(platform: string, architecture: string, source: string, products: string[]) {
+    // Create release key
+    const releaseKey = 'source-' + crypto.createHash('sha256').update(source).digest('hex');
+    const releaseInfo = {
+        name: releaseKey,
+        version: "1.0.0",
+        update: "",
+        isPrerelease: false
+    };
```

Comment:
What about looking for this info in `ProductFilesInfo.xml` (should exist when using an `mpm download` folder) or `VersionInfo.xml` (should exist when using a mounted ISO folder)? Then we could install system dependencies and place MATLAB in the right tool cache location. If we can't find either file (which would probably indicate a corrupt or incorrect source folder anyway), this approach could be a fallback. We do something similar in `run-matlab-command`.